driver/dma_control.c

```diff
@@ -83,38 +83,54 @@ struct dma_control {
 /* DMA. */
 
 
-static void reset_dma_controller(struct dma_control *dma)
+static int reset_dma_controller(struct dma_control *dma)
 {
     writel(CDMACR_Reset, &dma->regs->cdmacr);
 
     /* In principle we should wait for the reset to complete, though it doesn't
      * actually seem to take an observable time normally.  We use a deadline
      * just in case something goes wrong so we don't deadlock. */
-    unsigned long deadline = jiffies + 2;
+    unsigned long deadline = jiffies + msecs_to_jiffies(1);
     int count = 0;
-    while (readl(&dma->regs->cdmacr) & CDMACR_Reset  &&
+    int reset_stat = 0;
+    while ((reset_stat=readl(&dma->regs->cdmacr) & CDMACR_Reset)  &&
            time_before(jiffies, deadline))
         count += 1;
```

Comment:
I don't like assignment expressions at all.  Of course, in this case I agree that this is better than a `while(true) { ... if(...)break; }` loop.  In this particular case I like to use the following trick:
```c
    while (
        reset_stat = readl(&dma->regs->cdmacr),
        (reset_stat & CDMACR_Reset)  &&  time_before(jiffies, deadline))
    {
        count += 1;
    }
```
Here we use the comma operator to separate assignments from the test, and I've used brackets for the one-line body simply for layout.

Reply:
Done